lib/src/observable_map.dart

```diff
@@ -155,65 +155,37 @@ class ObservableMap<K, V> extends Observable implements Map<K, V> {
   String toString() => MapBase.mapToString(this);
 
   @override
-  // TODO: Dart 2.0 requires this method to be implemented.
-  // ignore: override_on_non_overriding_method
-  Map<K2, V2> cast<K2, V2>() {
-    throw new UnimplementedError("cast");
-  }
+  Map<K2, V2> cast<K2, V2>() => _map.cast<K2, V2>();
```

Comment:
`List.cast()` returns a `List<T>`, not `Iterable<T>`, so my impression is that the expected policy is that this returns an object of the receiver's own type, ObservableMap.

@lrhn, @leafpetersen thoughts?

Reply:
Probably true. I'd just wrap the current map here:
```dart
Map<K2, V2> cast<K2, V2>() {
  Map<Object, Object> self = this;
  return self is Map<K2, V2> ? self : this.retype<K2, V2>();
}
Map<K2, V2> retype<K2, V2>() => Map.castFrom<K, V, K2, V2>(this);
```

Reply:
So just to clarify, I _should_ return an `ObservableMap`?

Reply:
Done.

Reply:
I think so, yes. You wrap the *current* map, with all its behavior, at a new type. 

I do short-cuts in some cases where I know it won't make a difference, and to avoid cast-of-cast wrappers, but in general the default implementation is correct.